### DISCUSSION TITLE
The best practice of Github actions on a MonoRepo like repository

### DISCUSSION THREAD

**Discussion by hantsy** [community/community#25661] [Actions]
<p>I have some repositories which include more than one project, but I want to make every project apply its own build lifecycle, I created a standalone Github actions workflow file for each project.</p>
<p>For example, <a href="https://github.com/hantsy/spring-graphql-sample/tree/master/.github/workflows" class="inline-onebox" rel="noopener nofollow ugc">spring-graphql-sample/.github/workflows at master · hantsy/spring-graphql-sample · GitHub</a></p>
<p>In the workflow file, I used the <em>paths</em> attribute on push to differentiate the projects.</p>
<pre><code class="lang-auto">on:
  push:
    paths:
      - "dgs-client/**"
    branches: [ master ]
  pull_request:
    branches: [ master ]
#...
</code></pre>
<p>The problem is when pushing codes to <strong>master</strong> branch directly, it worked as expected, but when contributing codes via PRs, it will trigger all workflows.</p>
<p>I have to add another <code>paths</code> on the <code>pull_request</code> event ?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>Yes, you need to specify <code>paths</code> for the <code>pull_request</code> event as well. It’s actually the only other event that supports <code>paths</code>: <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a></p>
